test/ReachSets/unit_discretization.jl

```diff
@@ -4,92 +4,99 @@ let # preventing global scope
     δ = 0.01
 
-    # no bloating, do not use Pade approximation
-    discr_sys_homog = discretize(cont_sys_homog, δ, approx_model="nobloating", pade_expm=false)
+    # no bloating, use default options
+    discr_sys_homog = discretize(cont_sys_homog, δ, approximation="nobloating")
     @test inputdim(discr_sys_homog) == 0 # there is no input set!
 
     # no bloating, use Pade approximation
-    discr_sys_homog = discretize(cont_sys_homog, δ, approx_model="nobloating", pade_expm=true)
+    discr_sys_homog = discretize(cont_sys_homog, δ, approximation="nobloating", exp_method="pade")  
     @test inputdim(discr_sys_homog) == 0
 
-    # bloating, do not use Pade approximation
-    discr_sys_homog = discretize(cont_sys_homog, δ, pade_expm=false)
+    # bloating, default options
+    discr_sys_homog = discretize(cont_sys_homog, δ)
     @test inputdim(discr_sys_homog) == 0
 
     # bloating, first order
-    discr_sys_homog = discretize(cont_sys_homog, δ, approx_model="firstorder")
+    discr_sys_homog = discretize(cont_sys_homog, δ, approximation="firstorder")
+    @test inputdim(discr_sys_homog) == 0
+
+    # bloating, forward interpolation
+    discr_sys_homog = discretize(cont_sys_homog, δ, approximation="forward")
+    @test inputdim(discr_sys_homog) == 0
+
+    # bloating, backward interpolation
+    discr_sys_homog = discretize(cont_sys_homog, δ, approximation="backward")
     @test inputdim(discr_sys_homog) == 0
 
     # ===============================================================
     # Discretization of a continuous-time system with constant input
     # ===============================================================
-    U = Ball2(ones(4), 0.5)
-    cont_sys = ContinuousSystem(A, X0, U)
+    U = ConstantInput(Ball2(ones(4), 0.5))
+    B = Matrix{Float64}(1.0I, 4, 4)
+    cont_sys = IVP(CLCCS(A, B, nothing, U), X0)
 
-    # no bloating, do not use Pade approximation
-    discr_sys = discretize(cont_sys, δ, approx_model="nobloating", pade_expm=false)
+    # no bloating
+    discr_sys = discretize(cont_sys, δ, approximation="nobloating")
     @test inputdim(discr_sys) == 4
-    #@test length(discr_sys.U) == 1 # a constant input has no length method!
-    inputs = next_set(inputset(discr_sys)) # getter: inputset(discr_sys) = discr_sys.s.U
+
+    inputs = next_set(inputset(discr_sys))
     @test dim(inputs) == 4
-    @test isa(inputs, LinearMap)
+    @test isa(inputs, LazySets.LinearMap)
     @test isa(inputs.X, Ball2) && inputs.X.center == ones(4) && inputs.X.radius == 0.5
 
     # no bloating, use Pade approximation
-    discr_sys = discretize(cont_sys, δ, approx_model="nobloating", pade_expm=true)
+    discr_sys = discretize(cont_sys, δ, approximation="nobloating", exp_method="pade")
     @test inputdim(discr_sys) == 4
 
-    # bloating, do not use Pade approximation
-    discr_sys = discretize(cont_sys, δ, pade_expm=false)
+    # bloating, use scaling and squaring method
+    discr_sys = discretize(cont_sys, δ, exp_method="base")
```

Comment:
What is the "scaling and squaring method"?

Reply:
This is a standard method to compute the matrix exponential, and it is the one used in Julia base. It is the one used in Matlab's `expm` as well. See http://eprints.ma.man.ac.uk/634/1/high05e.pdf.

Reply:
I would say that "base" and "scaling and squaring" does not sound similar. I would have written "the base method" instead.